part3/benchmark.py

```
@@ -0,0 +1,223 @@
+                times: List[float] = []
+                x_hat: np.ndarray = np.array([])
+
+                for _ in range(5):
+                    start: float = time.perf_counter()
+                    try:
+                        with warnings.catch_warnings():
+                            warnings.simplefilter("ignore")
+                            curr_x_hat = solver(A_list, b_list)
+                        times.append(time.perf_counter() - start)
+                        x_hat = curr_x_hat
+                    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)